python-threatexchange/benchmarks/benchmark_vpdq_index.py

```diff
@@ -0,0 +1,158 @@
+    VPDQ_DISTANCE_THRESHOLD,
+)
+
+import typing as t
+
+from threatexchange.extensions.vpdq.vpdq_index import VPDQIndex
+from threatexchange.extensions.vpdq.video_vpdq import VideoVPDQSignal
+
+
+Oint = t.Optional[int]
+
+
+class IndexType(Enum):
+    BRUTE_FORCE = "brute_force"
+    FLAT = "flat"
+    SIGNAL_TYPE = "signal_type"
+
+    def __str__(self) -> str:
+        return self.value
+
+
+@contextmanager
+def timer(context: str, print_on_enter: bool = False):
+    if print_on_enter:
+        print(f"{context}...")
+    start = time.perf_counter()
+    end = start
+    yield lambda: end - start
+    end = time.perf_counter()
+    print(f"{context}: {end - start:.4f}s")
+
+
+def run_benchmark(
+    test_type: IndexType,
+    video_frames: int,
+    dataset_size: int,
+    query_size: int,
+):
+
+    data_generation_timer = nullcontext()
+    if video_frames * dataset_size > 10000:
+        data_generation_timer = timer("Generating data", True)
+    with data_generation_timer:
+        hashes = [get_random_VPDQs(video_frames) for _ in range(dataset_size)]
+    if test_type == IndexType.SIGNAL_TYPE:
+        build = lambda: build_signal(hashes)
+    elif test_type == IndexType.BRUTE_FORCE:
+        build = lambda: hashes
+    elif test_type == IndexType.FLAT:
+        build = lambda: build_flat(hashes)
+    else:
+        raise ValueError("Invalid test type")
+
+    with timer("build"):
+        index = build()
+
+    query_generation_timer = nullcontext()
+    if query_size > 10000:
+        query_generation_timer = timer("Generating queries", True)
+    with query_generation_timer:
+        hq = get_random_VPDQs(query_size)
+    if test_type == IndexType.SIGNAL_TYPE:
+        query = lambda: sigal_match(hq, index)
+    elif test_type == IndexType.BRUTE_FORCE:
+        query = lambda: brute_force_match(hq, index)
+    elif test_type == IndexType.FLAT:
+        query = lambda: index.search_with_distance_in_result(
+            hq, VPDQ_DISTANCE_THRESHOLD
+        )
+
+    with timer("query") as t:
+        query()
+    query_time = t()
+    print(f"  Per query: {1000 * query_time / query_size:.4f}ms")
+
+
+def build_flat(videos):
+    index = VPDQHashIndex()
+    for v in videos:
+        index.add_single_video(v)
+    return index
+
+
+def build_signal(hashes):
+    index = VPDQIndex()
+    for h in hashes:
+        index.add(vpdq_to_json(h), object())
+    return index
+
+
+def sigal_match(hash, index):
+    index.query(vpdq_to_json(hash))
+
+
+def brute_force_match(query, hashes):
+    for hash in hashes:
+        match_VPDQ_hash_brute(
+            query, hash, VPDQ_QUALITY_THRESHOLD, VPDQ_DISTANCE_THRESHOLD
+        )
+
+
+def get_argparse():
+    ap = argparse.ArgumentParser(description=__doc__)
+    ap.add_argument(
+        "--video-frames",
+        "-v",
+        type=int,
+        default=500,
+        help="How many frames in each video",
+    )
+    ap.add_argument(
+        "--dataset-size",
+        "-n",
```

Comment:
n corresponding the number of videos, and `-v` also existing might be confusing. Would suggest either:
1. Using `-n` for video and `-m` for frames
2. Using `-v` for videos and `-f` for average frames

Reply:
Took this option "Using -v for videos and -f for average frames"